tianshou/utils/net/common.py

```diff
@@ -9,35 +10,74 @@ class Net(nn.Module):
     """Simple MLP backbone. For advanced usage (how to customize the network),
     please refer to :ref:`build_the_network`.
 
-    :param concat: whether the input shape is concatenated by state_shape
+    :param bool concat: whether the input shape is concatenated by state_shape
         and action_shape. If it is True, ``action_shape`` is not the output
         shape, but affects the input shape.
+    :param bool dueling: whether to use dueling network to calculate Q values
+        (for Dueling DQN), defaults to False.
+    :param bool layer_norm: whether to use layer norm before ReLU, defaults to
+        False.
```

Comment:
For multi-layer perceptrons, batch normalization is more popular. Or maybe this can be a module so that users can use whatever they like (such as ``norm_layer=torch.nn.BatchNorm1d``)

Reply:
This is aligned with stable-baselines: https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/deepq/policies.py#L114
It's just a demo network, actually, I want users to create their own.

Reply:
`stable-baselines` is already dead. I don't know if it worth the effort to align with it.

Reply:
Yes, you are right lol

Reply:
> For multi-layer perceptrons, batch normalization is more popular. Or maybe this can be a module so that users can use whatever they like (such as norm_layer=torch.nn.BatchNorm1d)

This is indeed more popular in CV instead of RL and NLP. I previously use layer_norm more often.

Reply:
> I previously use layer_norm more often (because they are all sequential task).

I think RL tasks are sequential but because they are markovian, the decision itself is not sequential (only depend on the last state). What's more, I'm not proposing to use BN rather than LN, but proposing to modify the ``layer_norm `` argument so that this can support various nomalization layers (and you can still use layer norm as the default).

Reply:
Get it